### ISSUE TITLE
Tweaks to empty creator row behavior

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#4143]
After 2071a7c4cf1d80da4afb9cb6270ba422af6f5ca6

- If focus leaves an empty creator row (without the window itself losing focus), maybe it should go away? That wasn't the behavior in Z6, but it's a bit weirder now that the empty row can be in the middle of a list of creators. Also for something like pressing Esc — that just feels like it should make the empty row go away.
- Showing a field-mode button in unsaved rows in place of `+`, as suggested in https://github.com/zotero/zotero/pull/4096#issuecomment-2121337770, is a good idea. Not sure how weird that would look if we kept the `-`, though, and not sure which order we would put them in. (Maybe less weird with the mode switcher first, but then the `-` is misaligned.) Arguably `-` isn't really necessary if we remove the row whenever it loses focus or you press Esc? That's fairly similar to how all other rows work — you don't need to "close" field rows. But possible people would feel better with a `-` to click…
- Focus goes to the wrong place if you delete an empty row via the keyboard:

https://github.com/zotero/zotero/assets/783082/315d07ca-de34-4e37-9fc8-d1036bc17183

**Comment by abaevbog** [comment]
> Focus goes to the wrong place if you delete an empty row via the keyboard:

This, as well as a similar bug of a wrong component being focused on tab from a just-filled creator field, should be fixed in #4152

**Comment by dstillman** [comment]
> With this setup, if you open up a dropdown, it will not delete the unsaved row but I think it's better to be on the safer side to avoid deleting row when one doesn't expect it. Correct me if this setup sounds off.

No, that sounds perfect.

Is that an existing icon for the creator-mode toggle? Regardless, I don't think it's at all clear what that is, particularly in the + position. (It's also misaligned, but I don't think fixing that would help much.)

In any case, I think I was wrong and this doesn't actually make a lot of sense. This button just isn't really more relevant for empty rows than it is for existing rows. Ideally people aren't manually entering data to begin with, names from various sources can end up in the wrong mode, and being able to see the mode at a glance is [actually useful for support and troubleshooting](https://forums.zotero.org/discussion/comment/464065/#Comment_464065).

Also, while I know this screenshot is just an example, at a width like this it's of course pretty silly to not just show the toggle for all rows, since there's a massive amount of space. I still think we could consider what I mentioned in https://github.com/zotero/zotero/issues/3959#issuecomment-2088005512, conditionally showing the button based on the pane width. I'd even reframe it: we would be generally showing the button, only moving it into the menu if the pane was made really narrow.

Let's remove the button from here for now and merge the rest, and we can continue to think about that.

**Comment by abaevbog** [comment]
> Is that an existing icon for the creator-mode toggle?

Yes, this is our current [input-dual.svg](https://github.com/zotero/zotero/blob/main/chrome/skin/default/zotero/16/universal/input-dual.svg).

> Let's remove the button from here for now and merge the rest, and we can continue to think about that.

Ok, gotcha! I think that dynamically showing or hiding the buttons depending on the width would make a lot of sense and would also be fairly intuitive as well. I'll see what it could look like next. 

Meanwhile, I removed it from that PR and just kept the removing unsaved row on blur with a number of other smaller fixes to bugs that I recently encountered. It's a few edits mainly to how focus is handled (which gets a bit tricky when, say, "X more creators" label shows up) - let me know if it should be a pulled out into a separate PR. With all these numerous small behaviors in itemBox, adding a larger test suite to keep track of it would be great one day!
